### ISSUE TITLE
Missing some Security Header/DNS Security configurations.

### ISSUE THREAD

**Issue by c0nfigurati0n** [techlore/website#11]
Your website is missing the following Security Header/DNS Security configurations.

- [DNSSEC](https://en.wikipedia.org/wiki/Domain_Name_System_Security_Extensions)
- [HSTS](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security)
- [CSP](https://en.wikipedia.org/wiki/Content_Security_Policy)
- X XSS Protection ([Cross site scripting](https://en.wikipedia.org/wiki/Cross-site_scripting))
- X Content Type Options ([Content sniffing](https://en.wikipedia.org/wiki/Content_sniffing))
- [Referer header](https://developer.mozilla.org/enUS/docs/Web/Security/Referer_header:_privacy_and_security_concerns)

Seen as your website is missing some of these important Security Header/DNS Security configurations, i would suggest you look in to fixing these miss configurations and start implementing them asap.

This issue was opened by, 
Mr. Muffin,
Representative of [The New Oil](https://thenewoil.org/about.html),
**_(not opening this issue on behave of The New Oil)_**

**Full URLs:**
https://en.wikipedia.org/wiki/Domain_Name_System_Security_Extensions
https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security
https://en.wikipedia.org/wiki/Content_Security_Policy
https://en.wikipedia.org/wiki/Cross-site_scripting
https://en.wikipedia.org/wiki/Content_sniffing
https://developer.mozilla.org/enUS/docs/Web/Security/Referer_header:_privacy_and_security_concerns
https://thenewoil.org/about.html

**Comment by techlore-official** [comment]
Will look into these in the near future.

**Comment by MrMuffinIsTaken** [comment]
Alright, good to know you're going to look in to them.

**Comment by MrMuffinIsTaken** [comment]
@techlore-official here is the basic configuration for some of the header security options i mentioned.

DNSSEC
HSTS
CSP
X XSS Protection (Cross site scripting)
X Content Type Options (Content sniffing)
Referer header

Header Strict-Transport-Security: max-age=31536000; includeSubDomains
Header Content-Security-Policy: script-src 'self'
Header always set X-Frame-Options "SAMEORIGIN"
Header X-Content-Type-Options: nosniff
Header Referrer-Policy: no-referrer
Header always set Permissions-Policy "geolocation=(),midi=(),sync-xhr=(),microphone=(),camera=(),magnetometer=(),gyroscope=(),fullscreen=(self),payment=()"
Header Expect-CT: max-age=0, report-uri="https://{$subdomain}.report-uri.com/r/d/ct/reportOnly"
Header Cross-Origin-Embedder-Policy: (unsafe-none|require-corp); report-to="default"
Header Cross-Origin-Embedder-Policy-Report-Only: (unsafe-none|require-corp); report-to="default"
Header Cross-Origin-Opener-Policy: (same-origin|same-origin-allow-popups|unsafe-none); report-to="default"
Header Cross-Origin-Opener-Policy-Report-Only: (same-origin|same-origin-allow-popups|unsafe-none); report-to="default"
Header Cross-Origin-Resource-Policy: (same-site|same-origin|cross-origin)
Header X-XSS-Protection: 1; mode=block

I suggest you change it where necessary. Note that, if you're not planing to have any form of third party content on your website then you should be fine just to copy and past (if that's what you like to do) else change it where necessary (like i said before).

**Comment by MrMuffinIsTaken** [comment]
don't forget

**Comment by henryistaken** [comment]
DNSSEC is now configured. Looking into the others as time opens up!

**Comment by MrMuffinIsTaken** [comment]
Alright, thank you for listening! And taking this issue seriously. It goes to show us, that you care.

**Comment by henryistaken** [comment]
- As previously mentioned DNSSEC is good to go
- HSTS is now configured and I just submitted our preload submission to: https://hstspreload.org/
- X-Frame options & X-content-type options are configured
- Referer header is set
- Bonus: Permission header should be set as well

Only waiting on CSP. it's a bit more involved.

**Comment by MrMuffinIsTaken** [comment]
Yeah! I see. Thank you for keeping me in the loop! I also just want to mention the other following Security Headers/DNS Security options and configurations that you also maybe want to implement.

-  CAA -- [Certification Authority Authorization](https://en.wikipedia.org/wiki/DNS_Certification_Authority_Authorization) (DNS security option)
- MTA-STS -- [MTA Strict Transport Security](en.wikipedia.org/wiki/Simple_Mail_Transfer_Protocol#SMTP_MTA_Strict_Transport_Security) (email security option)
- TLS-RPT -- [SMTP TLS Reporting](https://en.wikipedia.org/wiki/Simple_Mail_Transfer_Protocol#SMTP_TLS_Reporting) (email security option)
- [Expect CT](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_ref-60) (reporting header)
- [X-XSS-Protection](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#cite_ref-76) -- Cross-site scripting (security header)
- [iodef](https://en.wikipedia.org/wiki/Incident_Object_Description_Exchange_Format) -- [Search for it on this page](https://en.wikipedia.org/wiki/Certification_Authority_Authorization) (cca reporting)

**Comment by jonaharagon** [comment]
@henryistaken:

- The CAA DNS record should be set to `0 issue "letsencrypt.org"`, but you'd have to check if orange website supports them.
- We could set up MTA-STS (which is like HSTS but for email), it just requires creating a subdomain to host a file, and I don't know how orange handles that.
- I don't think TLS-RPT provides value to us as we don't self-host email.
- `Expect-CT` is obsolete, because browsers already always expect CT.
- `X-XSS-Protection` should be **disabled**, i.e. `X-XSS-Protection = "0"` (https://github.com/privacyguides/privacyguides.org/pull/2021#issuecomment-1444083670)
- I don't think specifying a reporting email in the CAA record provides value to us.

This is not the _most_ secure, but a better-than-nothing CSP (copied from PG):

```
default-src 'none'; script-src 'self' 'unsafe-inline'; style-src 'self' 'unsafe-inline'; font-src 'self'; img-src data: 'self'; connect-src https://*.techlore.tech 'self'; frame-src https://www.youtube-nocookie.com; frame-ancestors 'self'; manifest-src 'self'
```

**Comment by henryistaken** [comment]
Looks like MTA-STS is properly configured. Will wait for @jonaharagon to double-check my work 🫡

**Comment by henryistaken** [comment]
![image](https://github.com/techlore/website/assets/64515030/52b4db78-61d5-4151-a8be-989ad7dd9d02)
CSP is good to go as well. Going to close this issue as I feel we've taken adequate steps for the moment. If anyone sees new issues/suggestions feel free to open a new issue.
